src/libsyntax/parse/attr.rs

```diff
@@ -130,12 +130,13 @@ impl<'a> Parser<'a> {
         };
 
         if inner_parse_policy == InnerAttributeParsePolicy::Permitted &&
-           self.token == token::Semi {
-            self.bump();
-            self.span_warn(span,
-                           "this inner attribute syntax is deprecated. The new syntax is \
-                            `#![foo]`, with a bang and no semicolon");
-            style = ast::AttrStyle::Inner;
+            self.token == token::Semi {
+                let token_str = self.this_token_to_string();
+                self.bump();
+                let msg = "unexpected semicolon";
+                let help_msg = format!("consider remove this semicolon: `{}`", &token_str);
```

Comment:
"remove" -> "removing"

Also it's better to use `span_label` here instead of `help` , so the "consider remove this semicolon" message appears closer to the error. Printing ": `{}`" would also be unnecessary in this case.
(There are many examples of using `span_label` in the parser.)

Reply:
thank for your advice.
I found that this situation was already covered by `expect item, found {}` like error suggestion just now, so I remove the whole extra `if {}` block here.

Reply:
Ha, indeed!